jujugui/static/gui/src/app/components/deployment/deployment.js

```diff
@@ -31,6 +31,39 @@ YUI.add('deployment-component', function() {
       getUnplacedUnitCount: React.PropTypes.func.isRequired,
     },
 
+    componentDidMount: function() {
```

Comment:
I may have misunderstood your question. If it is why do we do the check on `componentDidMount` and `componentWillReceiveProps` the answer would be in case for some reason we need to have a go-to-start button or something, I didn't want it to break.